src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/VSRestoreSettingsUtilities.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.ProjectModel;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class VSRestoreSettingsUtilities
+    {
+        private static string RestoreAdditionalProjectFallbackFolders = nameof(RestoreAdditionalProjectFallbackFolders);
+        private static string RestoreAdditionalProjectSources = nameof(RestoreAdditionalProjectSources);
```

Comment:
It would be less code/error prone to have a generic keyword such as ``|$Additional$|`` instead of one for each type. Then write a generic helper to split a list into 2 lists on that keyword.